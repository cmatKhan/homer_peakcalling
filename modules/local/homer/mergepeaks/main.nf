process HOMER_MERGEPEAKS {
    tag "$meta.id"
    label 'process_single'
    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'oras://community.wave.seqera.io/library/bioconductor-deseq2_bioconductor-edger_homer_samtools_pruned:9c603739ae7d4fd3'
        : 'community.wave.seqera.io/library/bioconductor-deseq2_bioconductor-edger_homer_samtools_pruned:08c7bb832e96c6bd'}"

    input:
    tuple val(meta), path(peaks) // peaks can be a list of peak files

    output:
    tuple val(meta), path("*_merged.txt"), emit: merged
    path "versions.yml"                  , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '5.1'
    
    """
    mergePeaks \\
        $args \\
        ${peaks.join(' ')} \\
        > ${prefix}_merged.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        homer: ${VERSION}
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '5.1'
    """
    touch ${prefix}_merged.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        homer: ${VERSION}
    END_VERSIONS
    """
}