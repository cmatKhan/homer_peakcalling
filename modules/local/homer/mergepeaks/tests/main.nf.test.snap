{
    "sarscov2 - bed - merge two peak files": {
        "content": [
            "merged_peaks_merged.txt",
            [
                "versions.yml:md5,5525de3ec5759617906fb41311dfb901"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-07T14:43:47.936609066"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "merged"
                        },
                        "merged_peaks_merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5525de3ec5759617906fb41311dfb901"
                ],
                "merged": [
                    [
                        {
                            "id": "merged"
                        },
                        "merged_peaks_merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5525de3ec5759617906fb41311dfb901"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-07T14:43:53.353525443"
    }
}